### PR TITLE
Functions for Collaborators

### PR DESCRIPTION
**Pull request by ethantkoenig**
Add functions corresponding to the following endpoints:
- `GET repos/:username/:reponame/collaborators`: list collaborators
- `GET repos/:username/:reponame/collaborators/:collaborator`: check if a user is a collaborator
- `DELETE repos/:username/:reponame/collaborators/:collaborator`: remove a collaborator

These endpoints are implemented in https://github.com/go-gitea/gitea/pull/375.

### CODE REVIEW

**Comment by bkcsoft** [comment]
LGTM

File: gitea/repo_collaborator.go [hunk 1]
```diff
@@ -10,6 +10,29 @@ import (
 	"fmt"
 )
 
+// ListCollaborators list a repository's collaborators
+func (c *Client) ListCollaborators(user, repo string) ([]*User, error) {
```
Comment (lunny) [inline gitea/repo_collaborator.go:14]: pager?

**Review by lunny — CHANGES_REQUESTED** [review]

File: gitea/repo_collaborator.go (hunk 1 above)
Comment (bkcsoft) [reply gitea/repo_collaborator.go:14]: this needs pagination yes, I must've missed that one 🙁

Comment (tboerger) [reply gitea/repo_collaborator.go:14]: IMHO pager makes only sense for potentially large lists

File: gitea/repo_collaborator.go [hunk 2]
```diff
@@ -10,6 +10,29 @@ import (
...
+	collaborators := make([]*User, 10)
```
Comment (lunny) [inline gitea/repo_collaborator.go]: ```collaborators := make([]*User, 0, 10)```

File: gitea/repo_collaborator.go (hunk 1 above)
Comment (lunny) [reply gitea/repo_collaborator.go:14]: That's OKay.

**Review by lunny — CHANGES_REQUESTED** [review]

File: gitea/repo_collaborator.go (hunk 1 above)
Comment (bkcsoft) [reply gitea/repo_collaborator.go:14]: @tboerger having 100 collaborators isn't large enough? :)

Comment (bkcsoft) [reply gitea/repo_collaborator.go:14]: and seeing as it's currently locked at 10 entries per call, pagination is required if you have more than that...

Comment (tboerger) [reply gitea/repo_collaborator.go:14]: 100 is nothing for an api ;)

Comment (ethantkoenig) [reply gitea/repo_collaborator.go:14]: @lunny @bkcsoft @tboerger Do we want pager or not? Happy to add one if we do, just want to make sure.

Comment (lunny) [reply gitea/repo_collaborator.go:14]: I agree pager is no need currently. But `collaborators := make([]*User, 0, 10)` is really not `collaborators := make([]*User, 10)`.

File: gitea/repo_collaborator.go (hunk 2 above)
Comment (ethantkoenig) [reply gitea/repo_collaborator.go]: Fixed.

**Review by lunny — APPROVED** [review]
LGTM

**Comment by tboerger** [comment]
LGTM

**Pull request merged**